### DISCUSSION TITLE
How to get folding working for mouse peasants such as myself?

### DISCUSSION THREAD

**Discussion by airtonix** [nvim-treesitter/nvim-treesitter#4009] [Q&A] [answered]
Unashamed mouse peasant from the gutter here `o7` 💀 

I'd like to know how to get code folding happening in the following way:

- little arrows in the gutter to indicate that something is folded ▶️ , or unfolded 🔽
- folding of code available for all lines based on logical blocks of code inferred from the AST that treesitter understands.
- clicking with my mouse on the little icons in the gutter folds/unfolds code.

Thanks for being so understanding and non judgemental of this humble mouse using gutter peasant.

**Comment by clason** [comment, marked as answer]
_Very_ off topic for this project, but see https://github.com/kevinhwang91/nvim-ufo.

**Reply by airtonix** [reply]
Exactly what i wanted !!!

```lua
local PluginSpec = {
    -- Cold Folding
    'kevinhwang91/nvim-ufo',
    dependencies = {
        'kevinhwang91/promise-async'
    },

    config = function()
        vim.o.foldcolumn = '2' -- '0' is not bad
        vim.o.foldlevel = 99 -- Using ufo provider need a large value, feel free to decrease the value
        vim.o.foldlevelstart = 99
        vim.o.foldenable = true

        require('ufo').setup({
            provider_selector = function(bufnr, filetype, buftype)
                return {'treesitter', 'indent'}
            end
        })
    end
}
return PluginSpec

```
![image](https://user-images.githubusercontent.com/61225/209242117-1ea21db1-9c6e-46e9-aafb-c57ffa8d47af.png)
